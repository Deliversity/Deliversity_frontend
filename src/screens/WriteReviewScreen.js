import React, {Component} from 'react';
import {View, StyleSheet, TextInput, TouchableOpacity} from 'react-native';
import {RadioButton} from 'react-native-paper';
import {Text} from 'native-base';
import axios from '../axiosConfig';
import {connect} from 'react-redux';
class WriteReviewScreen extends Component {
  static navigationOptions = {
    title: 'WriteReview',
  };
  constructor(props) {
    super(props);
    this.state = {
      score: null,
      comment: '',
    };
  }
  onClickSendReview = async () => {
    const data = {
      orderId: this.props.route.params.orderID,
      rating: this.state.score,
      content: this.state.comment,
    };
    if (this.props.user === '사용자') {
      await axios
        .post('/api/v1/order/review/rider', data)
        .then((res) => {
          alert('후기가 전달되었습니다.');
        })
        .catch((e) => {
          alert(e.response.data.message);
          this.props.navigation.goBack(null);
        });
    }
    if (this.props.user === '배달원') {
      await axios
        .post('/api/v1/order/review/user', data)
        .then((res) => {
          alert('후기가 전달되었습니다.');
        })
        .catch((e) => {
          alert(e.response.data.message);
          this.props.navigation.goBack(null);
        });
    }
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.box}>
          <Text style={styles.imageTitle}>
            {this.props.route.params.riderID}님의 점수를 매겨 주세요:)
          </Text>
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'center',
              paddingVertical: 15,
            }}>
            <View style={{flexDirection: 'column'}}>
              <Text style={styles.imageSubTitle}>1</Text>
              <RadioButton
                value="score"
                status={this.state.score === 1 ? 'checked' : 'unchecked'}
                onPress={() => this.setState({score: 1})}
              />
            </View>
            <View style={{flexDirection: 'column'}}>
              <Text style={styles.imageSubTitle}>2</Text>
              <RadioButton
                value="score"
                status={this.state.score === 2 ? 'checked' : 'unchecked'}
                onPress={() => this.setState({score: 2})}
              />
            </View>
            <View style={{flexDirection: 'column'}}>
              <Text style={styles.imageSubTitle}>3</Text>
              <RadioButton
                value="score"
                status={this.state.score === 3 ? 'checked' : 'unchecked'}
                onPress={() => this.setState({score: 3})}
              />
            </View>
            <View style={{flexDirection: 'column'}}>
              <Text style={styles.imageSubTitle}>4</Text>
              <RadioButton
                value="score"
                status={this.state.score === 4 ? 'checked' : 'unchecked'}
                onPress={() => this.setState({score: 4})}
              />
            </View>
            <View style={{flexDirection: 'column'}}>
              <Text style={styles.imageSubTitle}>5</Text>
              <RadioButton
                value="score"
                status={this.state.score === 5 ? 'checked' : 'unchecked'}
                onPress={() => this.setState({score: 5})}
              />
            </View>
          </View>
        </View>
        <View style={styles.box}>
          <Text style={styles.imageTitle}>코멘트 남기기</Text>
          <Text style={{fontSize: 10, marginBottom: 5}}>ex)매너가 좋아요</Text>
          <TextInput
            style={styles.textInputBox}
            value={this.state.comment}
            onChangeText={(text) => this.setState({comment: text})}
          />
        </View>
        <View
          style={{
            flex: 1,
            flexDirection: 'column',
            justifyContent: 'flex-end',
          }}>
          <TouchableOpacity
            onPress={() => {
              this.onClickSendReview();
            }}>
            <Text style={styles.panelButtonTitle}>후기 보내기</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  box: {
    borderBottomWidth: 1,
    borderBottomColor: '#fffaf0',
    flexDirection: 'column',
    paddingHorizontal: 20,
    paddingVertical: 20,
    marginBottom: 5,
    marginTop: 2,
    backgroundColor: 'white',
  },
  radiobox: {
    borderBottomWidth: 1,
    borderBottomColor: '#fffaf0',
    flexDirection: 'row',
    paddingHorizontal: 20,
    paddingVertical: 20,
    marginBottom: 5,
    marginTop: 2,
    backgroundColor: 'white',
    justifyContent: 'space-between',
  },
  imageSubTitle: {
    fontSize: 10,
    textAlign: 'center',
  },
  imageTitle: {
    fontWeight: 'bold',
    fontSize: 17,
    backgroundColor: 'transparent',
  },
  textInputBox: {
    width: '100%',
    height: 100,
    backgroundColor: '#fafad2',
    borderWidth: 3,
    borderColor: 'gray',
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    borderBottomLeftRadius: 15,
    borderBottomRightRadius: 15,
    marginTop: 10,
  },
  container: {
    flex: 1,
  },
  panelButtonTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
    backgroundColor: '#8fbc8f',
    width: '100%',
    textAlign: 'center',
  },
});

const mapStateToProps = (state) => ({
  user: state.authentication.user,
});
export default connect(mapStateToProps, {})(WriteReviewScreen);
