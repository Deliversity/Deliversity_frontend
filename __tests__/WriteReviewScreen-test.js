import 'react-native';

import React, {ReactElement} from 'react';
import WriteReviewScreen from '../src/screens/WriteReviewScreen';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
const mockStore = configureMockStore();
let props;
let component: ReactElement;
configure({adapter: new Adapter()});
beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
  // jest.mock('@react-native-firebase');
});

describe('[Temp] render', () => {
  let store;
  props = {
    route: {
      params: {
        idToken: 'randomToken',
      },
    },
  }; // fill test props
  beforeEach(() => {
    const initialState = {
      authentication: {user: '사용자'},
    };
    store = mockStore(initialState);
    component = shallow(
      <Provider store={store}>
        <WriteReviewScreen {...props} />
      </Provider>,
    );
  });
  it('renders without crashing', async () => {
    const renderResult = renderer.create(component);
    const renderedJson = renderResult.toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
