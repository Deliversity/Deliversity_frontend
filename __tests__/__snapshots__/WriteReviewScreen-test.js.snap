// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#fffaf0",
        "borderBottomWidth": 1,
        "flexDirection": "column",
        "marginBottom": 5,
        "marginTop": 2,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto",
            "fontSize": 16,
          },
          Object {
            "backgroundColor": "transparent",
            "fontSize": 17,
            "fontWeight": "bold",
          },
        ]
      }
    >
      님의 점수를 매겨 주세요:)
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          1
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 10,
                "borderWidth": 2,
                "height": 20,
                "margin": 8,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          2
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 10,
                "borderWidth": 2,
                "height": 20,
                "margin": 8,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          3
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 10,
                "borderWidth": 2,
                "height": 20,
                "margin": 8,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          4
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 10,
                "borderWidth": 2,
                "height": 20,
                "margin": 8,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Roboto",
                "fontSize": 16,
              },
              Object {
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          5
        </Text>
        <View
          accessibilityLiveRegion="polite"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 10,
                "borderWidth": 2,
                "height": 20,
                "margin": 8,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#fffaf0",
        "borderBottomWidth": 1,
        "flexDirection": "column",
        "marginBottom": 5,
        "marginTop": 2,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto",
            "fontSize": 16,
          },
          Object {
            "backgroundColor": "transparent",
            "fontSize": 17,
            "fontWeight": "bold",
          },
        ]
      }
    >
      코멘트 남기기
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Roboto",
            "fontSize": 16,
          },
          Object {
            "fontSize": 10,
            "marginBottom": 5,
          },
        ]
      }
    >
      ex)매너가 좋아요
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fafad2",
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "borderColor": "gray",
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "borderWidth": 3,
          "height": 100,
          "marginTop": 10,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Roboto",
              "fontSize": 16,
            },
            Object {
              "backgroundColor": "#8fbc8f",
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "width": "100%",
            },
          ]
        }
      >
        후기 보내기
      </Text>
    </View>
  </View>
</View>
`;
